native/components/message-markdown-rules.react.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import SimpleMarkdown from 'simple-markdown';
+
+type State = { [string]: any };
+
+type ASTNode = SingleASTNode | Array<SingleASTNode>;
+
+type SingleASTNode = {
+  type: string,
+  [string]: any,
+};
+
+type ReactElement = React$Element<any>;
+
+type Output<Result> = (node: ASTNode, state?: ?State) => Result;
+
+export default function(styles: any, opts: any = {}) {
+  const LINK_INSIDE = '(?:\\[[^\\]]*\\]|[^\\]]|\\](?=[^\\[]*\\]))*';
+  const LINK_HREF_AND_TITLE =
+    '\\s*<?([^\\s]*?)>?(?:\\s+[\'"]([\\s\\S]*?)[\'"])?\\s*';
+  var pressHandler = function(target) {
```

Comment:
1. Let's avoid `var` declarations in favor of `const`, or `let` if necessary
2. I think it's cleaner to use the newer closure syntax:

```js
const pressHandler = target => {
  [...]
};
```

(I'm mentioning this here once, but please make sure to apply it everywhere in the PR!)